devTools/phpstan-src.neon

```diff
@@ -1,9 +1,9 @@
 parameters:
     ignoreErrors:
-        - '#Access to an undefined property PhpParser\\Node(.*?)#'
-        - '#Call to an undefined method PhpParser\\Node(.*?)#'
         - '#expects .*, array<string>\|bool\|string\|null given#'
         - '#cast array<string>\|bool\|string\|null to#'
-        - '#Offset array<string>\|bool\|string\|null does not#'
-        - '#Possibly invalid array key type array<string>\|bool\|string\|null#'
-        - '#Access to undefined constant PhpParser\\NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN#'
+        - '#Return type \(void\) of method Infection\\Visitor\\.*::.*\(\) should be compatible with return type .* of method PhpParser\\Node.*::.*\(\)#'
+        -
+            path: '%currentWorkingDirectory%/src/Process/Runner/Parallel/ParallelProcessRunner.php'
```

Comment:
What is the line of this error? Trying to understand the reason..

Reply:
The left side of the || and the && are always true according to the inspector.
From what i can tell that is actually not the case, because it is running in paralel, meaning that things can change between the first and second check.

Reply:
Here's what PHPStan tells us:
```
 ------ --------------------------------- 
  Line   ParallelProcessRunner.php        
 ------ --------------------------------- 
  92     Left side of && is always true.  
  122    Left side of || is always true.  
 ------ --------------------------------- 
```
On line 92 we truly have a bogus situation, because if we remove that supposedly always-true check, `startProcess()` will try to unshift a process from an empty array, and fail. I could not understand why PHPStan decided it is so, so this could be a bug in PHPStan.

On the contrary, situation on the line 122 is different. A job is always either inside `$processesQueue` or `$currentProcesses`, so it may really make sense to remove that check from the left side of `||`. As far as I can tell E2E tests are okay with this change.